### PR TITLE
Release notes for edge-19.5.2

### PR DESCRIPTION
**Pull request by hawkw**
* CLI
  * Fixed `linkerd check` and `linkerd dashboard` failing when any control plane
    pod is not ready, even when multiple replicas exist (as in HA mode)
* Controller
  * Fixed control plane components failing on startup when the Kubernetes API
    returns an `ErrGroupDiscoveryFailed`
* Proxy
  * Added a dispatch timeout that limits the amount of time a request can be
    buffered in the proxy
  * Removed the limit on the number of concurrently active service discovery
    queries to the Destination service

Special thanks to @zaharidichev for adding end to end tests for proxies with
TLS!

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+* Controller
+  * Fixed control plane components failing on startup when the Kubernetes API
+    returns an `ErrGroupDiscoveryFailed`
+* Proxy
+  * Added a dispatch timeout that bounds the amount of time a request can be
+    buffered in the proxy
+  * Removed the limit on the number of concurrently active service discovery
+    to the Destination service
+* Internal
+  * Fixed potentially flaky assertions in the service profile integration tests
```
Comment (scottcarol) [inline CHANGES.md]: Question - is "potentially flaky" going to be understood by everyone? Would "Improved assertions in the service profile integration tests to avoid potential instability" be better? TIOLI

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
+## edge-19.5.2
+
+* CLI
+  * Fixed `linkerd check` and `linkerd dashboard` failing when any control plane
+    pod is not ready, even when multiple replicas exist (as in HA mode)
+* Controller
+  * Fixed control plane components failing on startup when the Kubernetes API
+    returns an `ErrGroupDiscoveryFailed`
+* Proxy
+  * Added a dispatch timeout that bounds the amount of time a request can be
```
Comment (scottcarol) [inline CHANGES.md]: TIOLI: "limits the amount of time" seems easier to understand to me instead of "bounds"

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+    buffered in the proxy
+  * Removed the limit on the number of concurrently active service discovery
```
Comment (scottcarol) [inline CHANGES.md]: Should this be "concurrently active service discoveries"?

Comment (hawkw) [reply CHANGES.md]: Ah, this was supposed to say "concurrently active service discovery queries" but the word "queries" disappeared somehow. Fixed it in e0971d9d247bdc68cbc35693b98efb376c8b02b4

**Comment by hawkw** [comment]
@scottcarol thanks for your suggestions! I tried to simplify the wording in f5d0b972a0c8c427823573e36dadbee1a75d5d34, what do you think?

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+    pod is not ready, even when multiple replicas exist (as in HA mode)
+* Controller
+  * Fixed control plane components failing on startup when the Kubernetes API
+    returns an `ErrGroupDiscoveryFailed`
+* Proxy
+  * Added a dispatch timeout that limits the amount of time a request can be
+    buffered in the proxy
+  * Removed the limit on the number of concurrently active service discovery
+    queries to the Destination service
+* Internal
```
Comment (olix0r) [inline CHANGES.md]: I'd prefer that we omit this section -- none of this is user-facing.

Comment (olix0r) [reply CHANGES.md]: We frequently just have a _Thanks ..._ at the bottom to call out contributions

Comment (hawkw) [reply CHANGES.md]: Okay, what do you think of 2dbc9fef3b7be911d7e46d570b21733c05ddec53?

**Comment by scottcarol** [comment]
👍 looks great to me!
